### PR TITLE
Valentin/add create member logic

### PR DESCRIPTION
**Pull request by ValentinNM**
ADDED: pulled main and staged changes
FIXED: error on setProfile & setAddress states and fetches
UPDATE: CreateMember.jsx working component

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -7,74 +7,116 @@ import ListOfMembers from "./components/ListOfMembers";
 import ListOfTrainers from "./components/ListOfTrainers";
 import CreateMembers from "./Pages/CreateMembers";
 import ViewMember from "./Pages/ViewMemeber"
-import EditMember from "./Pages/EditMember";
+import EditMember from "./Pages/EditMemberForm";
```
Comment (vherus) [inline src/App.js]: Although this is fine, the name of the component in the file is `EditMemberForm` - I'd recommend changing this to `import EditMemberForm` just for clarity with team members who might come to use it

File: src/App.js [hunk 2]
```diff
@@ -7,74 +7,116 @@ import ListOfMembers from "./components/ListOfMembers";
...
     trainers,
     classes,
-    members
+    members,
+    detailsToEdit,
+    contactEdit
   })
 
-  useEffect (() => {
+  useEffect(() => {
```
Comment (vherus) [inline src/App.js]: Using multiple useEffect hooks in a component is fine, but it's a little cleaner to stick to one where possible. A good solution here is to separate all your fetch calls into their own functions, then call each of the functions in the one useEffect. E.g.

```
function fetchClasses() {
    fetch(`${API_URL}/classes`)
      .then((res) => res.json())
      .then((classData) => {
        setClasses(classData);
      });
}

function fetchMembers() {}
function fetchTrainers() {}

useEffect(() => {
    fetchClasses();
    fetchMembers();
    fetchTrainers();
}, [])
```

File: src/Pages/CreateMembers.jsx [hunk 3]
```diff
@@ -4,30 +4,92 @@ import Box from "@mui/material/Box";
 import TextField from "@mui/material/TextField";
 
 export default function (props) {
```
Comment (vherus) [inline src/Pages/CreateMembers.jsx]: This component doesn't have a name which means it's an anonymous function. I'd recommend naming it `CreateMembers`

File: src/Pages/CreateMembers.jsx [hunk 4]
```diff
@@ -4,30 +4,92 @@ import Box from "@mui/material/Box";
...
+    membershipStatus: "",
+    picture: "smth",
+    firstName: "",
+    lastname: "",
+    houseNumber: 0,
+    streetName: "",
+    city: "",
+    postcode: "",
+    country: "",
+  });
```
Comment (vherus) [inline src/Pages/CreateMembers.jsx]: I can see what you're doing here but it's a little messy. I think adding some more state to represent `profile` and `address` would be great here:
`const [profileToCreate, setProfileToCreate] = useState({})`

Then when you get to the handle submit, you can change the `const newMember` to:

```
const newMember = {
...memberToCreate,
profile: {
  ...profileToCreate
  address: {
    ...addressToCreate
  }
}
}
```

File: src/Pages/EditClassForm.jsx [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+            .then((res) => res.json())
+            .then((classToUpdate) => {
+                console.log("Updated class", classToUpdate);
+
+                const updatedClass = classes.map((classe) => {
+                    if (classToUpdate.id === classe.id) {
+                        return classToUpdate;
+                    } else {
+                        return classe
+                    }
```
Comment (vherus) [inline src/Pages/EditClassForm.jsx]: You don't need to change this, but this if statement is a perfect opportunity to use a ternary (short-hand if) to bring this code block down to a single line:

```
const updatedClass = classes.map(classe => classToUpdate.id === classe.id ? classToUpdate : classe);
```

**Review by vherus — CHANGES_REQUESTED** [review]
The only change I'd like you to make is giving that anonymous component a name, you can choose whether to action the other comments. Good work!

File: src/App.js (hunk 2 above)
Comment (Emmanuel-Atama) [reply src/App.js]: I will effect this change and use just one useEffect

File: src/Pages/CreateMembers.jsx (hunk 3 above)
Comment (ValentinNM) [reply src/Pages/CreateMembers.jsx]: must have forgotten to update after adding more work after the template initialisation

**Review by vherus — APPROVED** [review]

**Comment by ValentinNM** [comment]
UPDATE: pulled from main + code refactor + cleaning up

**Pull request merged**